crates/cargo-wdk/tests/test_utils/mod.rs

```diff
@@ -191,3 +170,168 @@ where
          Please build using a Windows target."
     );
 }
+
+/// An RAII wrapper over a Win API named mutex
+pub struct NamedMutex {
+    handle: HANDLE,
+    // `ReleaseMutex` requires that it is called
+    // only by threads that own the mutex handle.
+    // Being `!Send` ensures that's always the case.
+    _not_send: PhantomData<*const ()>,
+}
+
+impl NamedMutex {
+    /// Acquires named mutex
+    pub fn acquire(name: &CStr) -> Result<Self, WinError> {
+        fn get_last_error() -> WinError {
+            // SAFETY: We have to just assume this function is safe to call
+            // because the windows crate has no documentation for it and
+            // the MSDN documentation does not specify any preconditions
+            // for calling it
+            unsafe { GetLastError().into() }
+        }
+
+        // SAFETY: The name ptr is valid because it comes from a CStr
+        let handle = unsafe { CreateMutexA(None, false, PCSTR(name.as_ptr().cast()))? };
+        if handle.is_invalid() {
+            return Err(get_last_error());
+        }
+
+        // SAFETY: The handle is valid since it was created right above
+        match unsafe { WaitForSingleObject(handle, INFINITE) } {
+            res if res == WAIT_OBJECT_0 || res == WAIT_ABANDONED => Ok(Self {
+                handle,
+                _not_send: PhantomData,
+            }),
+            _ => {
+                // SAFETY: The handle is valid since it was created right above
+                unsafe { CloseHandle(handle)? };
+                Err(get_last_error())
+            }
+        }
+    }
+}
+
+impl Drop for NamedMutex {
+    fn drop(&mut self) {
+        // SAFETY: the handle is guaranteed to be valid
+        // because this type itself created it and it
+        // was never exposed outside. Also the requirement
+        // that the calling thread must own the handle
+        // is upheld because this type is `!Send`
+        let _ = unsafe { ReleaseMutex(self.handle) };
+
+        // SAFETY: the handle is valid as explained above.
+        let _ = unsafe { CloseHandle(self.handle) };
+    }
+}
+
+/// Creates an [`std::process::Command`] object representing
+/// a cargo-wdk command invocation for tests.
+///
+/// It automatically locates the cargo-wdk binary and makes
+/// sure its environment variables are set correctly e.g.
+/// by removing those that might interfere with its operation.
+///
+/// # Arguments
+///
+/// * `cmd_name` - Name of the cargo-wdk command. Can be only "new" or "build"
+/// * `cmd_args` - Optional args for the command
+/// * `curr_working_dir` - Optional current working directory for the command
+pub fn create_cargo_wdk_cmd<P: AsRef<Path>>(
+    cmd_name: &str,
+    cmd_args: Option<&[&str]>,
+    curr_working_dir: Option<P>,
+    env_vars: Option<&[(&str, Option<&str>)]>,
```

Comment:
Nit: move this arg right after `cmd_args`. `cm_args` and `env_vars` look better together